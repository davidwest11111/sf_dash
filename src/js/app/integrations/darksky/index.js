import axios from 'axios';
import jsonpAdapter from 'axios-jsonp';


export const URL_BASE = 'https://api.darksky.net/forecast';

export default class DarkSky {
  static getUrl = ({
    // all params required
    base = URL_BASE,
    key,
    lat,
    long,
    exclude = 'minutely,flags,currently',
    lang = 'en',
    units = 'auto',
  }) => `${base}/${key}/${lat},${long}?exclude=${exclude}&lang=${lang}&units=${units}`;

  static async fetchWeatherData({ lat, long, key, units }) {
    // NOTE: this is a jsonp call. there are no status codes or errors to parse like with XHR
    // try/catch errors in components that call this, so just return the response as-is
    const response = await axios.get(this.getUrl({ lat, long, key, units }), {
      adapter: jsonpAdapter, // gets around CORS
    });
    return response;
  }

  static async fetchAllLocationsWeatherData(locations, key, units) {
    const darkSkyUnits = this.convertUnitsForDarkSky(units);
    const responses = await axios.all(
      locations.map(({ lat, long }) => (
        this.fetchWeatherData({ lat, long, key, units: darkSkyUnits })
      )),
    );
    return responses;
  }

  static isValidKey = async key => {
    const response = await this.fetchWeatherData({ lat: 38, long: -122, key });
    return !!response?.data;
  }

  static convertUnitsForDarkSky = (units = 'auto') => {
    switch (units) {
      case ('f'):
        return 'us';
      case ('c'):
        return 'si';
      case ('auto'):
        return 'auto';
      default:
        throw new Error(`Invalid Unit type: ${units}`);
    }
  };
}
