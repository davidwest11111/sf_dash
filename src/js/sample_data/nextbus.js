// The data returned is actually in XML, this is just what the X2JS library outputs

export const SAMPLE_PREDICTION = {
  direction: {
    prediction: [
      {
        _epochTime: '1588198229803',
        _seconds: '296',
        _minutes: '4',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6609',
        _block: '1620',
        _tripTag: '9359239',
        $$hashKey: 'object:19',
      },
      {
        _epochTime: '1588198681179',
        _seconds: '747',
        _minutes: '12',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6599',
        _block: '1622',
        _tripTag: '9359240',
        $$hashKey: 'object:20',
      }, { _epochTime: '1588199412858', _seconds: '1479', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6547', _block: '1633', _tripTag: '9359237', $$hashKey: 'object:21' }, { _epochTime: '1588199419316', _seconds: '1486', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6533', _block: '1628', _tripTag: '9359243', $$hashKey: 'object:22' }],
    _title: 'Inbound to Ferry Plaza',
  },
  message: [
    {
      _text: 'Service frequency for this stop is 10 minutes or less.',
      _priority: 'Normal',
    },
    {
      _text: 'Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info.',
      _priority: 'Low',
    }, { _text: 'Text COVID19SF to 888-777 for official updates. Visit sfdph.org', _priority: 'Low' }, { _text: 'Board thru rear doors. Front door for accessibility needs only.', _priority: 'Normal' }],
  _agencyTitle: 'SF Muni',
  _routeTitle: '14-Mission',
  _routeTag: '14',
  _stopTitle: 'Mission St & 24th St',
  _stopTag: '5565',
  $$hashKey: 'object:17',
  className: '',
};

// FIXME
export const SAMPLE_FAILED_PREDICTIONS = {

};

export const SAMPLE_PREDICTIONS_XML = `
<?xml version="1.0" encoding="utf-8" ?> 
<body copyright="All data copyright San Francisco Muni 2020.">
  <predictions agencyTitle="SF Muni" routeTitle="12-Folsom-Pacific" routeTag="12" stopTitle="Pacific Ave &amp; Grant Ave" stopTag="5837" dirTitleBecauseNoPredictions="Inbound to Van Ness">
    <message text="Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info." priority="Low"/>
    <message text="Text COVID19SF to 888-777 for official updates. Visit sfdph.org" priority="Low"/>
  </predictions>
  <predictions agencyTitle="SF Muni" routeTitle="14-Mission" routeTag="14" stopTitle="Mission St &amp; 24th St" stopTag="5565">
    <direction title="Inbound to Ferry Plaza">
      <prediction epochTime="1588394902295" seconds="11" minutes="0" isDeparture="false" dirTag="14___I_F00" vehicle="6591" block="1631" tripTag="9359290" />
      <prediction epochTime="1588395526282" seconds="635" minutes="10" isDeparture="false" dirTag="14___I_F00" vehicle="6539" block="1633" tripTag="9359291" />
    </direction>
    <message text="Service frequency for this stop is 10 minutes or less." priority="Normal"/>
    <message text="Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info." priority="Low"/>
    <message text="Text COVID19SF to 888-777 for official updates. Visit sfdph.org" priority="Low"/>
    <message text="Board thru rear doors. Front door for accessibility needs only." priority="Normal"/>
  </predictions>
</body>
`;
