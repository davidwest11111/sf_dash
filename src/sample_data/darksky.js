export const SAMPLE_HOURLY_FORECAST = {
  time: 1588276800,
  summary: 'Partly Cloudy',
  icon: 'partly-cloudy-day',
  precipIntensity: 0,
  precipProbability: 0.40, // this was hacked in for testing purposes
  temperature: 15.49,
  apparentTemperature: 15.49,
  dewPoint: 10.85,
  humidity: 0.74,
  pressure: 1019.9,
  windSpeed: 4.44,
  windGust: 5.76,
  windBearing: 263,
  cloudCover: 0.36,
  uvIndex: 7,
  visibility: 16.093,
  ozone: 319.6,
};

export const SAMPLE_DAILY_FORECAST = {
  time: 1588230000,
  summary: 'Partly cloudy throughout the day.',
  icon: 'partly-cloudy-day',
  sunriseTime: 1588252500,
  sunsetTime: 1588302060,
  moonPhase: 0.26,
  precipIntensity: 0.0089,
  precipIntensityMax: 0.031,
  precipIntensityMaxTime: 1588309200,
  precipProbability: 0.40, // this was hacked in for testing purposes
  precipType: 'rain',
  temperatureHigh: 16.23,
  temperatureHighTime: 1588285860,
  temperatureLow: 10.77,
  temperatureLowTime: 1588339020,
  apparentTemperatureHigh: 15.95,
  apparentTemperatureHighTime: 1588285860,
  apparentTemperatureLow: 11.04,
  apparentTemperatureLowTime: 1588339020,
  dewPoint: 8.85,
  humidity: 0.75,
  pressure: 1019.5,
  windSpeed: 4.22,
  windGust: 10.47,
  windGustTime: 1588295580,
  windBearing: 266,
  cloudCover: 0.41,
  uvIndex: 7,
  uvIndexTime: 1588278780,
  visibility: 16.093,
  ozone: 317.9,
  temperatureMin: 11.22,
  temperatureMinTime: 1588251120,
  temperatureMax: 16.23,
  temperatureMaxTime: 1588285860,
  apparentTemperatureMin: 11.49,
  apparentTemperatureMinTime: 1588251120,
  apparentTemperatureMax: 15.95,
  apparentTemperatureMaxTime: 1588285860,
};

export const SAMPLE_ALERT = {
  title: 'Flood Watch for Mason, WA',
  time: 1509993360,
  expires: 1510036680,
  description: '...FLOOD WATCH REMAINS IN EFFECT THROUGH LATE MONDAY NIGHT...\n',
  uri: 'http://alerts.weather.gov/cap/wwacapget.php?x=WA1255E4DB8494.FloodWatch.1255',
};


export const SAMPLE_LOCATION = {
  todaysForecast: {
    time: 1588230000,
    summary: 'Partly cloudy throughout the day.',
    icon: 'partly-cloudy-day',
    sunriseTime: 1588252500,
    sunsetTime: 1588302060,
    moonPhase: 0.26,
    precipIntensity: 0.0089,
    precipIntensityMax: 0.031,
    precipIntensityMaxTime: 1588309200,
    precipProbability: 0.4, // this was hacked in for testing purposes
    precipType: 'rain',
    temperatureHigh: 16.23,
    temperatureHighTime: 1588285860,
    temperatureLow: 10.77,
    temperatureLowTime: 1588339020,
    apparentTemperatureHigh: 15.95,
    apparentTemperatureHighTime: 1588285860,
    apparentTemperatureLow: 11.04,
    apparentTemperatureLowTime: 1588339020,
    dewPoint: 8.85,
    humidity: 0.75,
    pressure: 1019.5,
    windSpeed: 4.22,
    windGust: 10.47,
    windGustTime: 1588295580,
    windBearing: 266,
    cloudCover: 0.41,
    uvIndex: 7,
    uvIndexTime: 1588278780,
    visibility: 16.093,
    ozone: 317.9,
    temperatureMin: 11.22,
    temperatureMinTime: 1588251120,
    temperatureMax: 16.23,
    temperatureMaxTime: 1588285860,
    apparentTemperatureMin: 11.49,
    apparentTemperatureMinTime: 1588251120,
    apparentTemperatureMax: 15.95,
    apparentTemperatureMaxTime: 1588285860,
  },
  dailyForecasts: [
    { ...SAMPLE_DAILY_FORECAST },
    {
      time: 1588316400,
      summary: 'Clear throughout the day.',
      icon: 'clear-day',
      sunriseTime: 1588338840,
      sunsetTime: 1588388520,
      moonPhase: 0.3,
      precipIntensity: 0.0094,
      precipIntensityMax: 0.0454,
      precipIntensityMaxTime: 1588326900,
      precipProbability: 0.04,
      precipType: 'rain',
      temperatureHigh: 17.7,
      temperatureHighTime: 1588367220,
      temperatureLow: 12.58,
      temperatureLowTime: 1588430940,
      apparentTemperatureHigh: 17.42,
      apparentTemperatureHighTime: 1588367220,
      apparentTemperatureLow: 12.85,
      apparentTemperatureLowTime: 1588430940,
      dewPoint: 6.98,
      humidity: 0.63,
      pressure: 1019.5,
      windSpeed: 4.28,
      windGust: 9.14,
      windGustTime: 1588377660,
      windBearing: 276,
      cloudCover: 0.2,
      uvIndex: 8,
      uvIndexTime: 1588365360,
      visibility: 16.093,
      ozone: 343.3,
      temperatureMin: 10.77,
      temperatureMinTime: 1588339020,
      temperatureMax: 17.7,
      temperatureMaxTime: 1588367220,
      apparentTemperatureMin: 11.04,
      apparentTemperatureMinTime: 1588339020,
      apparentTemperatureMax: 17.42,
      apparentTemperatureMaxTime: 1588367220,
    },
    {
      time: 1588402800,
      summary: 'Overcast throughout the day.',
      icon: 'cloudy',
      sunriseTime: 1588425180,
      sunsetTime: 1588474920,
      moonPhase: 0.33,
      precipIntensity: 0.0188,
      precipIntensityMax: 0.0494,
      precipIntensityMaxTime: 1588418280,
      precipProbability: 0.1,
      precipType: 'rain',
      temperatureHigh: 16.18,
      temperatureHighTime: 1588458300,
      temperatureLow: 11.47,
      temperatureLowTime: 1588518000,
      apparentTemperatureHigh: 15.9,
      apparentTemperatureHighTime: 1588458300,
      apparentTemperatureLow: 11.74,
      apparentTemperatureLowTime: 1588518000,
      dewPoint: 10.11,
      humidity: 0.77,
      pressure: 1020.3,
      windSpeed: 3.38,
      windGust: 6.37,
      windGustTime: 1588463040,
      windBearing: 257,
      cloudCover: 0.87,
      uvIndex: 5,
      uvIndexTime: 1588452300,
      visibility: 15.997,
      ozone: 332.6,
      temperatureMin: 12.58,
      temperatureMinTime: 1588430940,
      temperatureMax: 16.18,
      temperatureMaxTime: 1588458300,
      apparentTemperatureMin: 12.85,
      apparentTemperatureMinTime: 1588430940,
      apparentTemperatureMax: 15.9,
      apparentTemperatureMaxTime: 1588458300,
    },
    {
      time: 1588489200,
      summary: 'Partly cloudy throughout the day.',
      icon: 'partly-cloudy-day',
      sunriseTime: 1588511520,
      sunsetTime: 1588561380,
      moonPhase: 0.37,
      precipIntensity: 0.0079,
      precipIntensityMax: 0.0203,
      precipIntensityMaxTime: 1588489200,
      precipProbability: 0.08,
      precipType: 'rain',
      temperatureHigh: 15.29,
      temperatureHighTime: 1588540440,
      temperatureLow: 10.28,
      temperatureLowTime: 1588593780,
      apparentTemperatureHigh: 15.01,
      apparentTemperatureHighTime: 1588540440,
      apparentTemperatureLow: 10.55,
      apparentTemperatureLowTime: 1588593780,
      dewPoint: 7.29,
      humidity: 0.68,
      pressure: 1021.5,
      windSpeed: 4.27,
      windGust: 11.61,
      windGustTime: 1588554480,
      windBearing: 273,
      cloudCover: 0.44,
      uvIndex: 8,
      uvIndexTime: 1588537860,
      visibility: 16.093,
      ozone: 327.1,
      temperatureMin: 11.43,
      temperatureMinTime: 1588519740,
      temperatureMax: 15.29,
      temperatureMaxTime: 1588540440,
      apparentTemperatureMin: 11.7,
      apparentTemperatureMinTime: 1588519740,
      apparentTemperatureMax: 15.01,
      apparentTemperatureMaxTime: 1588540440,
    },
    {
      time: 1588575600,
      summary: 'Clear throughout the day.',
      icon: 'clear-day',
      sunriseTime: 1588597860,
      sunsetTime: 1588647840,
      moonPhase: 0.41,
      precipIntensity: 0.0016,
      precipIntensityMax: 0.011,
      precipIntensityMaxTime: 1588647000,
      precipProbability: 0.01,
      precipType: 'rain',
      temperatureHigh: 18.18,
      temperatureHighTime: 1588633500,
      temperatureLow: 11.57,
      temperatureLowTime: 1588681860,
      apparentTemperatureHigh: 17.9,
      apparentTemperatureHighTime: 1588633500,
      apparentTemperatureLow: 11.84,
      apparentTemperatureLowTime: 1588681860,
      dewPoint: 6.4,
      humidity: 0.6,
      pressure: 1021.5,
      windSpeed: 3.33,
      windGust: 8.53,
      windGustTime: 1588638060,
      windBearing: 280,
      cloudCover: 0.01,
      uvIndex: 9,
      uvIndexTime: 1588622700,
      visibility: 16.093,
      ozone: 332.6,
      temperatureMin: 10.28,
      temperatureMinTime: 1588593780,
      temperatureMax: 18.18,
      temperatureMaxTime: 1588633500,
      apparentTemperatureMin: 10.55,
      apparentTemperatureMinTime: 1588593780,
      apparentTemperatureMax: 17.9,
      apparentTemperatureMaxTime: 1588633500,
    },
    {
      time: 1588662000,
      summary: 'Partly cloudy throughout the day.',
      icon: 'partly-cloudy-day',
      sunriseTime: 1588684200,
      sunsetTime: 1588734300,
      moonPhase: 0.45,
      precipIntensity: 0.0036,
      precipIntensityMax: 0.0072,
      precipIntensityMaxTime: 1588744800,
      precipProbability: 0.05,
      precipType: 'rain',
      temperatureHigh: 17.05,
      temperatureHighTime: 1588719120,
      temperatureLow: 12.25,
      temperatureLowTime: 1588767480,
      apparentTemperatureHigh: 16.77,
      apparentTemperatureHighTime: 1588719120,
      apparentTemperatureLow: 12.52,
      apparentTemperatureLowTime: 1588767480,
      dewPoint: 6.1,
      humidity: 0.58,
      pressure: 1021.6,
      windSpeed: 4.22,
      windGust: 10.83,
      windGustTime: 1588731060,
      windBearing: 292,
      cloudCover: 0.35,
      uvIndex: 5,
      uvIndexTime: 1588705380,
      visibility: 16.093,
      ozone: 329,
      temperatureMin: 11.57,
      temperatureMinTime: 1588681860,
      temperatureMax: 17.05,
      temperatureMaxTime: 1588719120,
      apparentTemperatureMin: 11.84,
      apparentTemperatureMinTime: 1588681860,
      apparentTemperatureMax: 16.77,
      apparentTemperatureMaxTime: 1588719120,
    },
  ],
  hourlyForecasts: [
    { ...SAMPLE_HOURLY_FORECAST },
    {
      time: 1588280400,
      summary: 'Clear',
      icon: 'clear-day',
      precipIntensity: 0,
      precipProbability: 0,
      temperature: 15.71,
      apparentTemperature: 15.71,
      dewPoint: 10.41,
      humidity: 0.71,
      pressure: 1019.7,
      windSpeed: 5.04,
      windGust: 7.32,
      windBearing: 262,
      cloudCover: 0.29,
      uvIndex: 7,
      visibility: 16.093,
      ozone: 319.9,
    },
    {
      time: 1588284000,
      summary: 'Partly Cloudy',
      icon: 'partly-cloudy-day',
      precipIntensity: 0,
      precipProbability: 0,
      temperature: 15.91,
      apparentTemperature: 15.91,
      dewPoint: 9.74,
      humidity: 0.67,
      pressure: 1019.4,
      windSpeed: 5.61,
      windGust: 8.8,
      windBearing: 259,
      cloudCover: 0.35,
      uvIndex: 6,
      visibility: 16.093,
      ozone: 320,
    },
    {
      time: 1588287600,
      summary: 'Clear',
      icon: 'clear-day',
      precipIntensity: 0.0051,
      precipProbability: 0.01,
      precipType: 'rain',
      temperature: 15.87,
      apparentTemperature: 15.87,
      dewPoint: 8.7,
      humidity: 0.62,
      pressure: 1019.6,
      windSpeed: 6.2,
      windGust: 9.54,
      windBearing: 280,
      cloudCover: 0.27,
      uvIndex: 4,
      visibility: 16.093,
      ozone: 320,
    },
  ],
  alerts: [
    { ...SAMPLE_ALERT }
  ],
  readable: 'Bernal Heights',
  lat: 37.7448205,
  long: -122.4100494,
};
