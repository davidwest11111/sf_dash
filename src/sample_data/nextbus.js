// The data returned is actually in XML, this is just what the X2JS library outputs

export const SAMPLE_PREDICTION = {
  direction: {
    prediction: [
      {
        _epochTime: '1588198229803',
        _seconds: '296',
        _minutes: '4',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6609',
        _block: '1620',
        _tripTag: '9359239',
        $$hashKey: 'object:19',
      },
      {
        _epochTime: '1588198681179',
        _seconds: '747',
        _minutes: '12',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6599',
        _block: '1622',
        _tripTag: '9359240',
        $$hashKey: 'object:20',
      }, { _epochTime: '1588199412858', _seconds: '1479', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6547', _block: '1633', _tripTag: '9359237', $$hashKey: 'object:21' }, { _epochTime: '1588199419316', _seconds: '1486', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6533', _block: '1628', _tripTag: '9359243', $$hashKey: 'object:22' }],
    _title: 'Inbound to Ferry Plaza',
  },
  message: [
    {
      _text: 'Service frequency for this stop is 10 minutes or less.',
      _priority: 'Normal',
    },
    {
      _text: 'Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info.',
      _priority: 'Low',
    }, { _text: 'Text COVID19SF to 888-777 for official updates. Visit sfdph.org', _priority: 'Low' }, { _text: 'Board thru rear doors. Front door for accessibility needs only.', _priority: 'Normal' }],
  _agencyTitle: 'SF Muni',
  _routeTitle: '14-Mission',
  _routeTag: '14',
  _stopTitle: 'Mission St & 24th St',
  _stopTag: '5565',
  $$hashKey: 'object:17',
  className: '',
};

// FIXME
export const SAMPLE_FAILED_PREDICTIONS = {

};
