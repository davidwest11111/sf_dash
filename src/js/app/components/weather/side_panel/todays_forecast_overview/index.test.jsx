import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import LocalStorage from 'app/utils/local_storage';
import { SAMPLE_DAILY_FORECAST } from 'sample_data/darksky';

import TodaysForecastOverview, { cToF, fToC } from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  test('localStorage weather units was not set', () => {
    wrapper = shallow((
      <TodaysForecastOverview todaysForecast={{ ...SAMPLE_DAILY_FORECAST }} />
    ));
  });

  test('localStorage weather units was set to c', () => {
    jest.spyOn(LocalStorage, 'get').mockReturnValue('c');

    wrapper = shallow((
      <TodaysForecastOverview todaysForecast={{ ...SAMPLE_DAILY_FORECAST }} />
    ));
  });

  test('localStorage weather units was set to f', () => {
    jest.spyOn(LocalStorage, 'get').mockReturnValue('f');

    wrapper = shallow((
      <TodaysForecastOverview todaysForecast={{ ...SAMPLE_DAILY_FORECAST }} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

test('cToF', () => {
  expect(cToF(22)).toEqual(71.6);
  expect(cToF(0)).toEqual(32);
  expect(cToF(40)).toEqual(104);
});

test('fToC', () => {
  expect(fToC(0)).toEqual(-17.77777777777778);
  expect(fToC(32)).toEqual(0);
  expect(fToC(73)).toEqual(22.77777777777778);
  expect(fToC(100)).toEqual(37.77777777777778);
});
