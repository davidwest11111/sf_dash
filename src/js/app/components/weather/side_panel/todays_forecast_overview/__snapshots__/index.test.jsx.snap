// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when localStorage weather units was not set 1`] = `
<div
  className="forecast mb-3 d-flex align-items-center justify-content-center"
>
  <WeatherIcon
    className=""
    icon="partly-cloudy-day"
  />
  <div
    className="text-truncate"
  >
    <div
      className="conditions text-truncate"
    >
      Partly cloudy throughout the day.
    </div>
    <div
      className="high-low"
    >
      11
      <span
        className="text-muted px-1"
      >
        /
      </span>
      16
      °
    </div>
  </div>
</div>
`;

exports[`outputs the expected tree when localStorage weather units was set to c 1`] = `
<div
  className="forecast mb-3 d-flex align-items-center justify-content-center"
>
  <WeatherIcon
    className=""
    icon="partly-cloudy-day"
  />
  <div
    className="text-truncate"
  >
    <div
      className="conditions text-truncate"
    >
      Partly cloudy throughout the day.
    </div>
    <div
      className="high-low"
    >
      11
      <span
        className="text-muted px-1"
      >
        /
      </span>
      16
      °
    </div>
    <div
      className="high-low-alt text-muted"
    >
      52
      /
      61
      °
      F
    </div>
  </div>
</div>
`;

exports[`outputs the expected tree when localStorage weather units was set to f 1`] = `
<div
  className="forecast mb-3 d-flex align-items-center justify-content-center"
>
  <WeatherIcon
    className=""
    icon="partly-cloudy-day"
  />
  <div
    className="text-truncate"
  >
    <div
      className="conditions text-truncate"
    >
      Partly cloudy throughout the day.
    </div>
    <div
      className="high-low"
    >
      11
      <span
        className="text-muted px-1"
      >
        /
      </span>
      16
      °
    </div>
    <div
      className="high-low-alt text-muted"
    >
      -12
      /
      -9
      °
      C
    </div>
  </div>
</div>
`;
